### ISSUE TITLE
Debug libraries is linked even though a configuration is Release when OpenCV and cmake are used

### ISSUE THREAD

**Issue by durswd** [microsoft/vcpkg#4601]
I use OpenCV with x64-windows-static through Cmake.
When I generates project files for VS 2017, debug libraries is linked even though a configuration is Release.

My vcpkg is placed on C:/Softwares/

- Additional Dependencies in Release mode

```
...
C:\Softwares\vcpkg\installed\x64-windows-static\debug\lib\jpeg.lib
C:\Softwares\vcpkg\installed\x64-windows-static\lib\libpng16.lib
C:\Softwares\vcpkg\installed\x64-windows-static\lib\tiff.lib
C:\Softwares\vcpkg\installed\x64-windows-static\debug\lib\lzma.lib
C:\Softwares\vcpkg\installed\x64-windows-static\lib\zlib.lib
C:\Softwares\vcpkg\installed\x64-windows-static\lib\ittnotify.lib
...
```

cmake command is as follows.

```
 cmake .. -G "Visual Studio 15 2017 Win64" "DCMAKE_TOOLCHAIN_FILE=C:\Softwares\vcpkg\scripts\buildsystems\vcpkg.cmake" -DVCPKG_TARGET_TRIPLET=x64-windows-static -DCMAKE_BUILD_TYPE=Release
```

cmake is as follows.

```
cmake_minimum_required(VERSION 3.8)
enable_language(CXX)

set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} /MT")

add_executable(CMakeTest main.cpp)

find_package(OpenCV REQUIRED)
target_include_directories(CMakeTest PRIVATE ${OpenCV_INCLUDE_DIRS})
target_link_libraries(CMakeTest PRIVATE ${OpenCV_LIBS})
```

**Comment by cramershoup** [comment]
I've encountered [similar issue with curl and cpr](https://github.com/Microsoft/vcpkg/issues/4606).
It seems only some libraries have this issue while some other works correctly.
In my case zlib was resolved correctly.
Hope this problem can be fixed.

**Comment by Neumann-A** [comment]
will be resolved by #5543
